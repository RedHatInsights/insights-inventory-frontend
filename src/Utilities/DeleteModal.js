import React from 'react';
import PropTypes from 'prop-types';
import {
  Button,
  ClipboardCopy,
  Icon,
  Level,
  LevelItem,
  Modal,
  Split,
  SplitItem,
  Stack,
  StackItem,
} from '@patternfly/react-core';

import ExclamationTriangleIcon from '@patternfly/react-icons/dist/js/icons/exclamation-triangle-icon';

const DeleteModal = ({
  handleModalToggle,
  isModalOpen,
  currentSytems,
  onConfirm,
}) => {
  let systemToRemove;
  let systemLabel = 'system';
  if (Array.isArray(currentSytems)) {
    systemToRemove =
      currentSytems.length === 1
        ? currentSytems[0].display_name
        : `${currentSytems.length} systems`;
    systemLabel = currentSytems.length === 1 ? systemLabel : 'systems';
  } else {
    systemToRemove = currentSytems.display_name;
  }

  return (
    <Modal
      variant="small"
      title="Delete from inventory"
      className="ins-c-inventory__table--remove sentry-mask data-hj-suppress"
      ouiaId="inventory-delete-modal"
      isOpen={isModalOpen}
      onClose={() => handleModalToggle(false)}
      appendTo={
        document.getElementsByClassName('inventory')[0] || document.body
      }
    >
      <Split hasGutter>
        <SplitItem>
          <Icon size="xl" status="warning">
            <ExclamationTriangleIcon />
          </Icon>
        </SplitItem>
        <SplitItem isFilled>
          <Stack hasGutter>
            <StackItem>
              {systemToRemove} will be removed from all {location.host}{' '}
              applications and services. You need to re-register the{' '}
              {systemLabel} to add it back to your inventory.
            </StackItem>
            <StackItem>
              To disable the daily upload for this {systemLabel}, use the
              following command:
            </StackItem>
            <StackItem>
              <ClipboardCopy>insights-client --unregister</ClipboardCopy>
            </StackItem>
          </Stack>
        </SplitItem>
      </Split>
      <Level hasGutter>
        <LevelItem>
          <Button
            variant="danger"
            ouiaId="confirm-inventory-delete"
            data-testid="confirm-inventory-delete"
            onClick={onConfirm}
          >
            Delete
          </Button>
          <Button
            variant="link"
            ouiaId="cancel-inventory-delete"
            onClick={() => handleModalToggle(false)}
          >
            Cancel
          </Button>
        </LevelItem>
      </Level>
    </Modal>
  );
};

const ActiveSystemProp = PropTypes.shape({
  id: PropTypes.string,
  displayName: PropTypes.string,
});

DeleteModal.propTypes = {
  isModalOpen: PropTypes.bool,
  currentSytems: PropTypes.oneOfType([
    ActiveSystemProp,
    PropTypes.arrayOf(ActiveSystemProp),
  ]),
  handleModalToggle: PropTypes.func,
  onConfirm: PropTypes.func,
};

DeleteModal.defaultProps = {
  isModalOpen: false,
  currentSytems: {},
  handleModalToggle: () => undefined,
  onConfirm: () => undefined,
};

export default DeleteModal;
