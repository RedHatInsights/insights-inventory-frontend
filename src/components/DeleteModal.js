import React from 'react';
import PropTypes from 'prop-types';
import {
    Modal,
    Split,
    SplitItem,
    Stack,
    StackItem,
    Level,
    LevelItem,
    Button,
    ClipboardCopy
} from '@patternfly/react-core';
import { ExclamationTriangleIcon } from '@patternfly/react-icons';
import { FormattedMessage } from 'react-intl';
import { defaultMessages } from '@redhat-cloud-services/frontend-components-translations';
import messages from '../messages';

const DeleteModal = ({ handleModalToggle, isModalOpen, currentSytem, onConfirm }) => (
    <Modal
        isSmall
        title="Remove from inventory"
        className="ins-c-inventory__table--remove"
        isOpen={isModalOpen}
        onClose={() => handleModalToggle(false)}
    >
        <Split gutter="md">
            <SplitItem><ExclamationTriangleIcon size="xl" className="ins-m-alert" /></SplitItem>
            <SplitItem isFilled>
                <Stack gutter="md">
                    <StackItem>
                        <FormattedMessage
                            {...messages.systemRemoveInfo}
                            values={{
                                systemToRemove: Array.isArray(currentSytem) ?
                                    currentSytem.length === 1 ?
                                        currentSytem[0].display_name :
                                        currentSytem.length :
                                    currentSytem.displayName,
                                host: location.host,
                                system: Array.isArray(currentSytem) ? currentSytem.length : 1
                            }}
                        />
                    </StackItem>
                    <StackItem>
                        <FormattedMessage
                            {...messages.disableDailyUploads}
                            values={{
                                system: Array.isArray(currentSytem) ? currentSytem.length : 1
                            }}
                        />
                    </StackItem>
                    <StackItem>
                        <ClipboardCopy>insights-client --unregister</ClipboardCopy>
                    </StackItem>
                </Stack>
            </SplitItem>
        </Split>
        <Level gutter="md">
            <LevelItem>
                <Button variant="danger" onClick={onConfirm}>
                    <FormattedMessage
                        {...defaultMessages.remove}
                    />
                </Button>
                <Button variant="link" onClick={() => handleModalToggle(false)}>
                    <FormattedMessage
                        {...defaultMessages.cancel}
                    />
                </Button>
            </LevelItem>
        </Level>
    </Modal>
);

const ActiveSystemProp = PropTypes.shape({
    id: PropTypes.string,
    displayName: PropTypes.string
});

DeleteModal.propTypes = {
    isModalOpen: PropTypes.bool,
    intl: PropTypes.shape({
        formatMessage: PropTypes.func
    }),
    currentSytem: PropTypes.oneOfType([ActiveSystemProp, PropTypes.arrayOf(ActiveSystemProp)]),
    handleModalToggle: PropTypes.func,
    onConfirm: PropTypes.func
};

DeleteModal.defaultProps = {
    isModalOpen: false,
    currentSytem: {},
    handleModalToggle: () => undefined,
    onConfirm: () => undefined
};

export default DeleteModal;
