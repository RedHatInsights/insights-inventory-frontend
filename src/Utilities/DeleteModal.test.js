/* eslint-disable camelcase */
import React from 'react';
import DeleteModal from './DeleteModal';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

describe('EntityTable', () => {
  describe('DOM', () => {
    it('should render correctly with no system', () => {
      const view = render(<DeleteModal />);

      expect(view.baseElement).toMatchSnapshot();
    });

    it('should render correctly with one system', () => {
      const view = render(
        <DeleteModal
          currentSytems={{ display_name: 'something' }}
          isModalOpen
        />
      );

      screen.getByText(
        /something will be removed from all localhost:5000 applications and services/i
      );
      expect(view.baseElement).toMatchSnapshot();
    });

    it('should render correctly with multiple systems', () => {
      const view = render(
        <DeleteModal
          currentSytems={[
            { display_name: 'something' },
            { display_name: 'another' },
          ]}
          isModalOpen
        />
      );

      screen.getByText(
        /2 systems will be removed from all localhost:5000 applications and services/i
      );
      expect(view.baseElement).toMatchSnapshot();
    });
  });

  describe('API', () => {
    afterEach(() => {
      jest.clearAllMocks();
    });

    it('should call close on X click', async () => {
      const onClose = jest.fn();
      render(
        <DeleteModal
          currentSytems={[
            { display_name: 'something' },
            { display_name: 'another' },
          ]}
          isModalOpen
          handleModalToggle={onClose}
        />
      );

      await userEvent.click(
        screen.getByRole('button', {
          name: /close/i,
        })
      );
      expect(onClose).toHaveBeenCalled();
    });

    it('should call close on cancel click', async () => {
      const onClose = jest.fn();
      render(
        <DeleteModal
          currentSytems={[
            { display_name: 'something' },
            { display_name: 'another' },
          ]}
          isModalOpen
          handleModalToggle={onClose}
        />
      );

      await userEvent.click(
        screen.getByRole('button', {
          name: /cancel/i,
        })
      );
      expect(onClose).toHaveBeenCalled();
    });

    it('should call onConfirm', async () => {
      const onConfirm = jest.fn();
      render(
        <DeleteModal
          currentSytems={[
            { display_name: 'something' },
            { display_name: 'another' },
          ]}
          isModalOpen
          onConfirm={onConfirm}
        />
      );

      await userEvent.click(
        screen.getByRole('button', {
          name: /remove/i,
        })
      );
      expect(onConfirm).toHaveBeenCalled();
    });
  });
});
