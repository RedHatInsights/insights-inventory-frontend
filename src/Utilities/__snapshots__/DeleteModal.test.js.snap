// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntityTable DOM should render correctly with multiple systems 1`] = `
<body
  class="pf-c-backdrop__open"
>
  <div
    aria-hidden="true"
  />
  <div>
    <div
      class="pf-c-backdrop"
    >
      <div
        class="pf-l-bullseye"
      >
        <div
          aria-describedby="pf-modal-part-4"
          aria-labelledby="pf-modal-part-3"
          aria-modal="true"
          class="pf-c-modal-box ins-c-inventory__table--remove sentry-mask data-hj-suppress pf-m-sm"
          data-ouia-component-id="inventory-delete-modal"
          data-ouia-component-type="PF4/ModalContent"
          data-ouia-safe="true"
          id="pf-modal-part-2"
          role="dialog"
        >
          <button
            aria-disabled="false"
            aria-label="Close"
            class="pf-c-button pf-m-plain"
            data-ouia-component-id="inventory-delete-modal-ModalBoxCloseButton"
            data-ouia-component-type="PF4/Button"
            data-ouia-safe="true"
            type="button"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              height="1em"
              role="img"
              style="vertical-align: -0.125em;"
              viewBox="0 0 352 512"
              width="1em"
            >
              <path
                d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
              />
            </svg>
          </button>
          <header
            class="pf-c-modal-box__header"
          >
            <h1
              class="pf-c-modal-box__title"
              id="pf-modal-part-3"
            >
              
              <span
                class="pf-c-modal-box__title-text"
              >
                Remove from inventory
              </span>
            </h1>
          </header>
          <div
            class="pf-c-modal-box__body"
            id="pf-modal-part-4"
          >
            <div
              class="pf-l-split pf-m-gutter"
            >
              <div
                class="pf-l-split__item"
              >
                <svg
                  aria-hidden="true"
                  class="ins-m-alert"
                  fill="currentColor"
                  height="3em"
                  role="img"
                  style="vertical-align: -0.375em;"
                  viewBox="0 0 576 512"
                  width="3em"
                >
                  <path
                    d="M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
                  />
                </svg>
              </div>
              <div
                class="pf-l-split__item pf-m-fill"
              >
                <div
                  class="pf-l-stack pf-m-gutter"
                >
                  <div
                    class="pf-l-stack__item"
                  >
                    2 systems
                     will be removed from all 
                    localhost:5000
                     
                    applications and services. You need to re-register the
                     
                    systems
                     to add it back to your inventory.
                  </div>
                  <div
                    class="pf-l-stack__item"
                  >
                    To disable the daily upload for this 
                    systems
                    , use the following command:
                  </div>
                  <div
                    class="pf-l-stack__item"
                  >
                    <div
                      class="pf-c-clipboard-copy"
                      data-ouia-component-type="PF4/ClipboardCopy"
                      data-ouia-safe="true"
                    >
                      <div
                        class="pf-c-clipboard-copy__group"
                      >
                        <input
                          aria-invalid="false"
                          aria-label="Copyable input"
                          class="pf-c-form-control"
                          data-ouia-component-id="OUIA-Generated-TextInputBase-2"
                          data-ouia-component-type="PF4/TextInput"
                          data-ouia-safe="true"
                          id="text-input-1"
                          type="text"
                          value="insights-client --unregister"
                        />
                        <button
                          aria-disabled="false"
                          aria-label="Copy to clipboard"
                          aria-labelledby="copy-button-1 text-input-1"
                          class="pf-c-button pf-m-control"
                          data-ouia-component-id="OUIA-Generated-Button-control-2"
                          data-ouia-component-type="PF4/Button"
                          data-ouia-safe="true"
                          id="copy-button-1"
                          type="button"
                        >
                          <svg
                            aria-hidden="true"
                            fill="currentColor"
                            height="1em"
                            role="img"
                            style="vertical-align: -0.125em;"
                            viewBox="0 0 448 512"
                            width="1em"
                          >
                            <path
                              d="M320 448v40c0 13.255-10.745 24-24 24H24c-13.255 0-24-10.745-24-24V120c0-13.255 10.745-24 24-24h72v296c0 30.879 25.121 56 56 56h168zm0-344V0H152c-13.255 0-24 10.745-24 24v368c0 13.255 10.745 24 24 24h272c13.255 0 24-10.745 24-24V128H344c-13.2 0-24-10.8-24-24zm120.971-31.029L375.029 7.029A24 24 0 0 0 358.059 0H352v96h96v-6.059a24 24 0 0 0-7.029-16.97z"
                            />
                          </svg>
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="pf-l-level pf-m-gutter"
            >
              <div>
                <button
                  aria-disabled="false"
                  class="pf-c-button pf-m-danger"
                  data-ouia-component-id="confirm-inventory-delete"
                  data-ouia-component-type="PF4/Button"
                  data-ouia-safe="true"
                  type="button"
                >
                  Remove
                </button>
                <button
                  aria-disabled="false"
                  class="pf-c-button pf-m-link"
                  data-ouia-component-id="cancel-inventory-delete"
                  data-ouia-component-type="PF4/Button"
                  data-ouia-safe="true"
                  type="button"
                >
                  Cancel
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`EntityTable DOM should render correctly with no system 1`] = `
<body>
  <div />
  <div />
</body>
`;

exports[`EntityTable DOM should render correctly with one system 1`] = `
<body
  class="pf-c-backdrop__open"
>
  <div
    aria-hidden="true"
  />
  <div>
    <div
      class="pf-c-backdrop"
    >
      <div
        class="pf-l-bullseye"
      >
        <div
          aria-describedby="pf-modal-part-3"
          aria-labelledby="pf-modal-part-2"
          aria-modal="true"
          class="pf-c-modal-box ins-c-inventory__table--remove sentry-mask data-hj-suppress pf-m-sm"
          data-ouia-component-id="inventory-delete-modal"
          data-ouia-component-type="PF4/ModalContent"
          data-ouia-safe="true"
          id="pf-modal-part-1"
          role="dialog"
        >
          <button
            aria-disabled="false"
            aria-label="Close"
            class="pf-c-button pf-m-plain"
            data-ouia-component-id="inventory-delete-modal-ModalBoxCloseButton"
            data-ouia-component-type="PF4/Button"
            data-ouia-safe="true"
            type="button"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              height="1em"
              role="img"
              style="vertical-align: -0.125em;"
              viewBox="0 0 352 512"
              width="1em"
            >
              <path
                d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
              />
            </svg>
          </button>
          <header
            class="pf-c-modal-box__header"
          >
            <h1
              class="pf-c-modal-box__title"
              id="pf-modal-part-2"
            >
              
              <span
                class="pf-c-modal-box__title-text"
              >
                Remove from inventory
              </span>
            </h1>
          </header>
          <div
            class="pf-c-modal-box__body"
            id="pf-modal-part-3"
          >
            <div
              class="pf-l-split pf-m-gutter"
            >
              <div
                class="pf-l-split__item"
              >
                <svg
                  aria-hidden="true"
                  class="ins-m-alert"
                  fill="currentColor"
                  height="3em"
                  role="img"
                  style="vertical-align: -0.375em;"
                  viewBox="0 0 576 512"
                  width="3em"
                >
                  <path
                    d="M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
                  />
                </svg>
              </div>
              <div
                class="pf-l-split__item pf-m-fill"
              >
                <div
                  class="pf-l-stack pf-m-gutter"
                >
                  <div
                    class="pf-l-stack__item"
                  >
                    something
                     will be removed from all 
                    localhost:5000
                     
                    applications and services. You need to re-register the
                     
                    system
                     to add it back to your inventory.
                  </div>
                  <div
                    class="pf-l-stack__item"
                  >
                    To disable the daily upload for this 
                    system
                    , use the following command:
                  </div>
                  <div
                    class="pf-l-stack__item"
                  >
                    <div
                      class="pf-c-clipboard-copy"
                      data-ouia-component-type="PF4/ClipboardCopy"
                      data-ouia-safe="true"
                    >
                      <div
                        class="pf-c-clipboard-copy__group"
                      >
                        <input
                          aria-invalid="false"
                          aria-label="Copyable input"
                          class="pf-c-form-control"
                          data-ouia-component-id="OUIA-Generated-TextInputBase-1"
                          data-ouia-component-type="PF4/TextInput"
                          data-ouia-safe="true"
                          id="text-input-0"
                          type="text"
                          value="insights-client --unregister"
                        />
                        <button
                          aria-disabled="false"
                          aria-label="Copy to clipboard"
                          aria-labelledby="copy-button-0 text-input-0"
                          class="pf-c-button pf-m-control"
                          data-ouia-component-id="OUIA-Generated-Button-control-1"
                          data-ouia-component-type="PF4/Button"
                          data-ouia-safe="true"
                          id="copy-button-0"
                          type="button"
                        >
                          <svg
                            aria-hidden="true"
                            fill="currentColor"
                            height="1em"
                            role="img"
                            style="vertical-align: -0.125em;"
                            viewBox="0 0 448 512"
                            width="1em"
                          >
                            <path
                              d="M320 448v40c0 13.255-10.745 24-24 24H24c-13.255 0-24-10.745-24-24V120c0-13.255 10.745-24 24-24h72v296c0 30.879 25.121 56 56 56h168zm0-344V0H152c-13.255 0-24 10.745-24 24v368c0 13.255 10.745 24 24 24h272c13.255 0 24-10.745 24-24V128H344c-13.2 0-24-10.8-24-24zm120.971-31.029L375.029 7.029A24 24 0 0 0 358.059 0H352v96h96v-6.059a24 24 0 0 0-7.029-16.97z"
                            />
                          </svg>
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="pf-l-level pf-m-gutter"
            >
              <div>
                <button
                  aria-disabled="false"
                  class="pf-c-button pf-m-danger"
                  data-ouia-component-id="confirm-inventory-delete"
                  data-ouia-component-type="PF4/Button"
                  data-ouia-safe="true"
                  type="button"
                >
                  Remove
                </button>
                <button
                  aria-disabled="false"
                  class="pf-c-button pf-m-link"
                  data-ouia-component-id="cancel-inventory-delete"
                  data-ouia-component-type="PF4/Button"
                  data-ouia-safe="true"
                  type="button"
                >
                  Cancel
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
